awscrt/mqtt.py

```diff
@@ -499,11 +499,14 @@ def subscribe(self, topic, qos, callback=None):
             def callback_wrapper(topic, payload, dup, qos, retain):
                 try:
                     callback(topic=topic, payload=payload, dup=dup, qos=QoS(qos), retain=retain)
-                except TypeError:
-                    # This callback used to have fewer args.
-                    # Try again, passing only those those args, to cover case where
-                    # user function failed to take forward-compatibility **kwargs.
-                    callback(topic=topic, payload=payload)
+                except TypeError as e:
+                    if "got an unexpected keyword argument" in str(e) and callback.__name__ in str(e):
```

Comment:
Is there a more robust way to handle this? What if someone changes the string value of the error? I don't think they would expect that to affect people's code execution beyond logging.